models/actions/task_output.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package actions
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+
+	"code.gitea.io/gitea/models/db"
+)
+
+// ActionTaskOutput represents an output of ActionTask.
+// So the outputs are bound to a task, that means when a completed job has been rerun,
+// the outputs of the job will be reset because the task is new.
+// It's by design, to avoid the outputs of the old task to be mixed with the new task.
+type ActionTaskOutput struct {
+	ID      int64
+	TaskID  int64  `xorm:"INDEX UNIQUE(task_id_key)"`
+	Key     string `xorm:"VARCHAR(255)"`
+	KeyHash string `xorm:"CHAR(40) UNIQUE(task_id_key)"`
+	Value   string `xorm:"TEXT"`
```

Comment:
Personally I would suggest to use two words .... one word conflicts a lot , like this one https://github.com/go-gitea/gitea/pull/23630#issuecomment-1515720495

It would waste a lot of time debugging and might cause bug if some code is not covered, there were a lot of `user` related bugs in history.

And I think it's worth to write to the backend guideline .......

Reply:
Fixed in 9eb634e039 .

TBH, `ActionTaskOutput.OutputKey` smalls bad to me. But I can't think of a better word and maybe it's worth to help debugging.

Reply:
I can see that the `hash` column is used for `unique`?

If most keys are short keys, I think using `ID` +  `Key` is better than `ID` + `KeyHash`, unless most keys are long keys (longer than the hash length).

For example:

* `ID` +  `Key`, then database stores:
    *  123, key-1
    *  123, key-2
    *  123, key-3 

* `ID` +  `KeyHash`, then database stores:
    *  123, (32 or 40 chars)
    *  123, (32 or 40 chars)
    *  123, (32 or 40 chars)


The `ID` +  `Key` seems more friendly to database indices if most keys are short.

Reply:
That's my opinion, if you think it's bad, maybe we could have some deep discussion? What about others think .....